lib/src/importer-registry.ts

```diff
@@ -7,10 +7,19 @@ import {URL} from 'url';
 import {inspect} from 'util';
 
 import * as utils from './utils';
-import {FileImporter, Importer, Options} from './vendor/sass';
+import {
+  FileImporter,
+  Importer,
+  NodePackageImporter,
+  Options,
+} from './vendor/sass';
 import * as proto from './vendor/embedded_sass_pb';
 import {catchOr, thenOr, PromiseOr} from './utils';
 
+export const nodePackageImporter: NodePackageImporter = {
+  _NodePackageImporterBrand: '',
```

Comment:
Maybe this should use a Symbol rather than an empty string (external code can create a property with an empty string as value but won't be able to create an object with the same symbol as value without reusing the exported object)

Reply:
I'm open to making this value something else, but I think a Symbol would essentially add a limited deterrent, but not give us any additional assurance. The [check currently](https://github.com/sass/embedded-host-node/pull/260/files#diff-7f23a509904ce33b4c02b582e976900ecfd6f440b3ccbd08a7e8e7eb87e9ecbeR67-R68) is already able to differentiate the exact exported object from a user-created object that conforms to the NodePackageImporter type.

Reply:
The runtime check does. But the type checker would not if you create an object with such property.
when using a symbol, you won't be able to create an object with the same symbol (unless the symbol gets exported to allow that)

Reply:
I think a solution for that is to use the Symbol as key as well:

```ts
// sass exports
declare const NPISymbol: unique symbol;
export interface NodePackageImporterSymbolKey {
  [NPISymbol]: unknown;
}

export const nodePackageImporterSymbolKey: NodePackageImporterSymbolKey = {
  [NPISymbol]: NPISymbol
};
```

there, there is no way to access the symbol from the exported `sass.nodePackageImporter` (except by using `Object.getOwnPropertySymbols` and then guessing which symbol is the right one in that list, but the type checker will still not allow you to refer to that symbol to defined the type for this `fakeSymbolImporterTyped`).

Reply:
I've updated the [Codesandbox](https://codesandbox.io/s/sass-package-importer-type-exploration-2r4sjy?file=/src/index.ts), and I think that still can be minimally evaded with a cast-

```ts
// sass exports
export const compileSymbolKey = (pkgImporter: NodePackageImporterSymbolKey) =>
  null;

// sass usage
const fakeSymbolKey = {} as sass.NodePackageImporterSymbolKey;
sass.compileSymbolKey(fakeSymbolKey);
```

My thought is that each of these are adding higher barriers to jump, more of a sign to a developer that they should not be doing this. If more deterrent is needed, an alternate tact would be to make it more explicit to the developer that they are doing something wrong, something like-

```ts
export const NodePackageImporterWarning: NodePackageImporterWarning = {
  _NodePackageImporterBrand:
    "Do not make your own instance. You must use the exported object."
};
```

This would be easier to fake, but harder for developers to expect it to work.

I'll defer to @nex3 on how much of a deterrent we want to make this.

Reply:
@jamesnw any time you use the `as` operator in typescript, you basically disable the type checker on that line (well, only partially as typescript still expects some overlap between the inferred type and the cast to reject full mistakes, requiring an intermediate cast to `unknown` if this is intended, but I could do `const fakeSymbolKey = true as unknown as sass.NodePackageImporterSymbolKey;` and be accepted).
You cannot both rely on the type checker to report invalid usages and telling it to assume that your code is valid...

Reply:
I'm not _too_ worried about people working around the type checking here, but I think the `declare const symbol` approach more accurately communicates the intent anyway so I'm in favor of it.

Reply:
I've made this change, with the spec change [here](https://github.com/sass/sass/pull/3748).

Reply:
ah, using the symbol itself as `sass.nodePackageImporter` is indeed even better